front/src/applications/stdcmV2/components/StdcmConsist.tsx

```diff
@@ -0,0 +1,154 @@
+import { useAppDispatch } from 'store';
+
+import StdcmCard from './StdcmCard';
+import StdcmSuggestions from './StdcmSuggestions';
+
+interface StdcmSuggestionsConsistOption
+  extends SelectOptionObject,
+    Omit<LightRollingStockWithLiveries, 'id'> {
+  value: LightRollingStockWithLiveries;
+}
+
+function ConsistCardTitle({
+  rollingStock,
+}: {
+  rollingStock?: LightRollingStockWithLiveries | null;
+}) {
+  if (!rollingStock) return null;
+
+  return (
+    <div className="consist-img w-50 d-flex justify-content-end">
+      <RollingStock2Img rollingStock={rollingStock} />
+    </div>
+  );
+}
+
+export default function StdcmConsist({ isPending = false }: { isPending?: boolean }) {
+  const { speedLimitByTag, speedLimitsByTags, dispatchUpdateSpeedLimitByTag } =
+    useStoreDataForSpeedLimitByTagSelector();
+
+  const { updateRollingStockID } = useOsrdConfActions() as StdcmConfSliceActions;
+  const dispatch = useAppDispatch();
+
+  // TODO: Handle error
+  const { data: { results: rollingStocks } = { results: [] }, isSuccess } =
+    enhancedEditoastApi.endpoints.getLightRollingStock.useQuery({
+      pageSize: 1000,
+    });
+
+  const { getRollingStockID } = useOsrdConfSelectors() as StdcmConfSelectors;
+  const rollingStockID = useSelector(getRollingStockID);
+  const selectedRs = rollingStocks.find((rs) => rs.id === rollingStockID);
+
+  const [filteredRollingStockList, setFilteredRollingStockList] =
+    useState<LightRollingStockWithLiveries[]>(rollingStocks);
+  const [isSelectChanged, setIsSelectChanged] = useState(false);
+
+  const { filters, searchMateriel } = useFilterRollingStock({
+    isSuccess,
+    rollingStocks,
+    filteredRollingStockList,
+    setFilteredRollingStockList,
+  });
+
+  const getLabel = (rs: LightRollingStockWithLiveries) => {
+    let res = '';
+    const { metadata, name } = rs;
+
+    const series = metadata?.series ?? (metadata?.reference || '');
+    const subseries =
+      metadata?.series && metadata.series !== metadata.subseries
+        ? metadata.subseries
+        : metadata?.detail || '';
+
+    if (series) res += series;
+    if (subseries) res += series ? ` (${subseries})` : subseries;
+    if (name) res += ` -- ${name}`;
+    return res;
+  };
+
+  const onInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    searchMateriel(e.target.value);
+  };
+
+  const onInputOnBlur = () => {
+    // avoid to set tractionEngineId to null if the onBlur is due to a select change
+    if (!isSelectChanged) {
+      if (filteredRollingStockList.length === 1) {
+        dispatch(updateRollingStockID(filteredRollingStockList[0].id));
+      } else {
+        dispatch(updateRollingStockID(undefined));
+      }
+    }
+    setIsSelectChanged(false);
+  };
+
+  const onSelectSuggestion = (option: StdcmSuggestionsConsistOption) => {
+    dispatch(updateRollingStockID(option.value.id));
+    setIsSelectChanged(true);
+  };
+
+  useEffect(() => {
```

Comment:
There is a bug in useStoreDataForRollingStockSelector, can you update what you return ? If rollingStockId is undefined, you should also return undefined for the rollingStock (and not the previous rs selected)